frb_codegen/src/generator/dart/mod.rs

```diff
@@ -474,16 +507,26 @@ fn generate_api_fill_to_wire_func(ty: &IrType, ir_file: &IrFile) -> String {
     }
 }
 
-fn generate_wire2api_func(ty: &IrType, ir_file: &IrFile) -> String {
-    let body = TypeDartGenerator::new(ty.clone(), ir_file).wire2api_body();
+fn generate_wire2api_func(ty: &IrType, ir_file: &IrFile, dart_api_class_name: &str) -> String {
+    let bridge_requirement = if let StructRef(IrTypeStructRef { name, freezed: _ }) = ty {
+        if has_methods(name, ir_file) {
+            format!("{} bridge,", dart_api_class_name)
+        } else {
+            "".to_string()
+        }
+    } else {
+        "".to_string()
+    };
```

Comment:
nit: 

```
if matches!(ty, StructRef(IrTypeStructRef{name}) if has_methodds(name, irfile)) { format!(..) } else {""}

Reply:
Does `matches!` allow for `if`? 
`
matches!(ty, StructRef(IrTypeStructRef{name, ..}) if has_methods(name, irfile) { format!("{} bridge,", dart_api_class_name) } else {"".to_string()});`

says

`no rules expected the token `{``

Reply:
`if matches!(ty, something if somethingelse) { sth } else {sth}`

https://doc.rust-lang.org/std/macro.matches.html